payments-core/src/test/java/com/stripe/android/payments/core/authentication/Stripe3DS2AuthenticatorTest.kt

```diff
@@ -0,0 +1,399 @@
+package com.stripe.android.payments.core.authentication
+
+import android.app.Activity
+import android.content.Context
+import androidx.activity.result.ActivityResultLauncher
+import androidx.test.core.app.ApplicationProvider
+import com.google.common.truth.Truth
+import com.google.common.truth.Truth.assertThat
+import com.nhaarman.mockitokotlin2.KArgumentCaptor
+import com.nhaarman.mockitokotlin2.any
+import com.nhaarman.mockitokotlin2.anyOrNull
+import com.nhaarman.mockitokotlin2.argWhere
+import com.nhaarman.mockitokotlin2.argumentCaptor
+import com.nhaarman.mockitokotlin2.eq
+import com.nhaarman.mockitokotlin2.mock
+import com.nhaarman.mockitokotlin2.never
+import com.nhaarman.mockitokotlin2.verify
+import com.nhaarman.mockitokotlin2.whenever
+import com.stripe.android.ApiKeyFixtures
+import com.stripe.android.PaymentAuthConfig
+import com.stripe.android.PaymentRelayStarter
+import com.stripe.android.StripePaymentController
+import com.stripe.android.model.PaymentIntentFixtures
+import com.stripe.android.model.Stripe3ds2AuthResultFixtures
+import com.stripe.android.model.Stripe3ds2AuthResultFixtures.ARES_CHALLENGE_FLOW
+import com.stripe.android.model.Stripe3ds2Fingerprint
+import com.stripe.android.model.Stripe3ds2FingerprintTest
+import com.stripe.android.model.Stripe3ds2Fixtures
+import com.stripe.android.model.StripeIntent
+import com.stripe.android.networking.AnalyticsEvent
+import com.stripe.android.networking.AnalyticsRequest
+import com.stripe.android.networking.AnalyticsRequestExecutor
+import com.stripe.android.networking.AnalyticsRequestFactory
+import com.stripe.android.networking.ApiRequest
+import com.stripe.android.networking.StripeRepository
+import com.stripe.android.payments.PaymentFlowResult
+import com.stripe.android.stripe3ds2.service.StripeThreeDs2Service
+import com.stripe.android.stripe3ds2.transaction.ChallengeParameters
+import com.stripe.android.stripe3ds2.transaction.MessageVersionRegistry
+import com.stripe.android.stripe3ds2.transaction.SdkTransactionId
+import com.stripe.android.stripe3ds2.transaction.Stripe3ds2ActivityStarterHost
+import com.stripe.android.stripe3ds2.transaction.Transaction
+import com.stripe.android.view.AuthActivityStarter
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.runBlocking
+import kotlinx.coroutines.test.TestCoroutineDispatcher
+import kotlinx.coroutines.test.runBlockingTest
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.RobolectricTestRunner
+import java.util.UUID
+
+@RunWith(RobolectricTestRunner::class)
+@ExperimentalCoroutinesApi
+class Stripe3DS2AuthenticatorTest {
+    private val testDispatcher = TestCoroutineDispatcher()
+    private val activity: Activity = mock()
+    private val host = AuthActivityStarter.Host.create(activity)
+
+    private val sdkTransactionId = mock<SdkTransactionId>().also {
+        whenever(it.value).thenReturn(UUID.randomUUID().toString())
+    }
```

Comment:
Why mock this instead of creating an actual object?

Reply:
oh didn't realize i'm reimplementing it :P changed to the actual object